### PR TITLE
Introduce search by using xpath syntax.

### PR DESCRIPTION
**Pull request by 4383**
User can now retrieve elements by using xpath syntax like.

Also introduce some minor changes on unit tests part to
simplify tox using and to avoid some misconfigurations.

Related to #21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,13 @@
 
 Get data from yaml file directly in your shell
 
+Niet is like sed for JSON/YAML data - you can use it to slice and filter and
```
Comment (dj4ngo) [inline README.md]: I do not agree with sed... sed can do really a lot of things and don't take care about languages, it manage only text .... 
 You could compare it to xmllint or jq (for json) but with the capacity to deal with yaml, json AND xml.
Here we can add/remove/edit som keys, but not so much compared to sed ...

File: README.md [hunk 2]
```diff
@@ -70,15 +76,22 @@ Fizz
 Buzz
 $ echo '{"foo": "bar", "fizz": {"buzz": ["1", "2", "Fizz", "4", "Buzz"]}}' | niet fizz.buzz -f squote
 '1' '2''Fizz' '4' 'Buzz'
-$ echo '{"foo": "bar", "fizz": {"buzz": ["1", "2", "Fizz", "4", "Buzz"]}}' | niet . -f yaml
+$ echo '{"foo": "bar", "fizz": {"buzz": ["1", "2", "fizz", "4", "buzz"]}}' | niet . -f yaml
```
Comment (dj4ngo) [inline README.md]: If you want ..... lol I do not understand why changing casse, but whatever

File: README.md [hunk 3]
```diff
@@ -107,7 +120,7 @@ $ niet ".project.meta.name" /path/to/your/file.yaml
 my-project
 $ niet ".project.foo" /path/to/your/file.yaml
 bar
-$ niet ".project.list-items" /path/to/your/file.yaml
+$ niet ".project.list" /path/to/your/file.yaml
```
Comment (dj4ngo) [inline README.md]: could be great to have a different commit for changing list-items to list, it a bit annoying to review ^^

Comment (4383) [reply README.md]: No it sound like a bug so in a first time I prefer to simply transform these examples to search a little bit more in few days to eventually fix this behaviour...

File: README.md (hunk 2 above)
Comment (4383) [reply README.md]: me too lol I don't remember why...
I guess is due to the following output or something like that...

By the way it's not an issue

Comment (dj4ngo) [reply README.md]: no, not at all ;)

File: README.md (hunk 3 above)
Comment (dj4ngo) [reply README.md]: no, this is not a bug ^^ It's just an example in documentation .... ^^
But whatever, no problem

Comment (4383) [reply README.md]: I guess using a dash with jsmesg on a search introduce some errors... I need to check why jsmesg react like this and I guessing submit a fix to jmesg to avoid this behaviour.
